LibGit2Sharp.NotFoundException: Object not found - no match for id

Any idea why I would get this error and is there anyway to ignore the error? The error is LibGit2Sharp.NotFoundException: Object not found - no match for id

Hi,
Not sure it's the same cause, but I got the same exception.
Here is the full stacktrace:
` ERROR [10/21/16 15:44:39:77] An unexpected error occurred:
 LibGit2Sharp.NotFoundException: Object not found - no match for id (5a8f976b3c528351cef3d7f632721bfdc785e9c)
    at LibGit2Sharp.Core.Ensure.HandleError(Int32 result)
    at LibGit2Sharp.Core.Proxy.git_revwalk_next(RevWalkerHandle walker)
    at LibGit2Sharp.CommitLog.CommitEnumerator.MoveNext()
    at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
    at GitVersion.VersionCalculation.FallbackBaseVersionStrategy.<GetVersions>d__0.MoveNext()
    at System.Linq.Enumerable.<SelectManyIterator>d__16`2.MoveNext()
    at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
    at GitVersion.VersionCalculation.BaseVersionCalculator.GetBaseVersion(GitVersionContext context)
    at GitVersion.VersionCalculation.NextVersionCalculator.FindVersion(GitVersionContext context)
    at GitVersion.GitVersionFinder.FindVersion(GitVersionContext context)
    at GitVersion.ExecuteCore.<>c__DisplayClass6_0.<ExecuteInternal>b__0(IRepository repo)
    at GitVersion.GitPreparer.WithRepository[TResult](Func`2 action)
    at GitVersion.ExecuteCore.ExecuteGitVersion(String targetUrl, String dynamicRepositoryLocation, Authentication authentication, String targetBranch, Boolean noFetch, String workingDirectory, String commitId, Config overrideConfig)
    at GitVersion.SpecifiedArgumentRunner.Run(Arguments arguments, IFileSystem fileSystem)
    at GitVersion.Program.VerifyArgumentsAndRun()
`
I git this error while testing the new Shallow clone in VSTS:


And then at the end of the script, I got:
```
##[error]System.Exception: Unexpected exit code 1 returned from tool GitVersion.exe
    at Microsoft.TeamFoundation.DistributedTask.Task.Internal.InvokeToolCmdlet.ProcessRecord()
    at System.Management.Automation.CommandProcessor.ProcessRecord()
 ##[error]PowerShell script completed with 1 errors.
```

I think in case of shallow clone (with an option?) you should ignore this errors

--- a/src/GitVersionCore/Configuration/GitVersionConfigurationException.cs
+++ b/src/GitVersionCore/Configuration/GitVersionConfigurationException.cs
@@ -1,19 +1,12 @@
 namespace GitVersion
 {
     using System;
-    using System.Runtime.Serialization;
 
     [Serializable]
     public class GitVersionConfigurationException : GitVersionException
     {
         public GitVersionConfigurationException(string msg)
             : base(msg)
-        {
-        }
-
-
-        protected GitVersionConfigurationException(SerializationInfo info, StreamingContext context)
-            : base(info, context)
         {
         }
     }

--- a/src/GitVersionCore/Configuration/GitVersionConfigurationException.cs
+++ b/src/GitVersionCore/Configuration/GitVersionConfigurationException.cs
@@ -1,10 +1,19 @@
 namespace GitVersion
 {
     using System;
+    using System.Runtime.Serialization;
 
-    public class GitVersionConfigurationException : Exception
+    [Serializable]
+    public class GitVersionConfigurationException : GitVersionException
     {
-        public GitVersionConfigurationException(string msg) : base(msg)
+        public GitVersionConfigurationException(string msg)
+            : base(msg)
+        {
+        }
+
+
+        protected GitVersionConfigurationException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
         {
         }
     }

--- a/src/GitVersionCore/Configuration/OldConfigurationException.cs
+++ b/src/GitVersionCore/Configuration/OldConfigurationException.cs
@@ -4,15 +4,16 @@ namespace GitVersion
     using System.Runtime.Serialization;
 
     [Serializable]
-    public class OldConfigurationException : Exception
+    public class OldConfigurationException : GitVersionException
     {
-        public OldConfigurationException(string message) : base(message)
+        public OldConfigurationException(string message)
+            : base(message)
         {
         }
 
-        protected OldConfigurationException(
-            SerializationInfo info,
-            StreamingContext context) : base(info, context)
+
+        protected OldConfigurationException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
         {
         }
     }

--- a/src/GitVersionCore/Configuration/OldConfigurationException.cs
+++ b/src/GitVersionCore/Configuration/OldConfigurationException.cs
@@ -1,19 +1,12 @@
 namespace GitVersion
 {
     using System;
-    using System.Runtime.Serialization;
 
     [Serializable]
     public class OldConfigurationException : GitVersionException
     {
         public OldConfigurationException(string message)
             : base(message)
-        {
-        }
-
-
-        protected OldConfigurationException(SerializationInfo info, StreamingContext context)
-            : base(info, context)
         {
         }
     }

--- a/src/GitVersionCore/GitVersionCore.csproj
+++ b/src/GitVersionCore/GitVersionCore.csproj
@@ -118,6 +118,7 @@
     <Compile Include="GitVersionCache.cs" />
     <Compile Include="GitVersionCacheKey.cs" />
     <Compile Include="GitVersionCacheKeyFactory.cs" />
+    <Compile Include="GitVersionException.cs" />
     <Compile Include="Helpers\EncodingHelper.cs" />
     <Compile Include="Helpers\FileSystem.cs" />
     <Compile Include="Helpers\IFileSystem.cs" />

--- a/src/GitVersionCore/GitVersionException.cs
+++ b/src/GitVersionCore/GitVersionException.cs
@@ -1,0 +1,31 @@
+﻿namespace GitVersion
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    [Serializable]
+    public class GitVersionException : ApplicationException
+    {
+        public GitVersionException()
+        {
+        }
+
+
+        public GitVersionException(string message)
+            : base(message)
+        {
+        }
+
+
+        public GitVersionException(string message, Exception innerException)
+            : base(message, innerException)
+        {
+        }
+
+
+        protected GitVersionException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+        }
+    }
+}

--- a/src/GitVersionCore/GitVersionException.cs
+++ b/src/GitVersionCore/GitVersionException.cs
@@ -1,16 +1,12 @@
 ﻿namespace GitVersion
 {
     using System;
-    using System.Runtime.Serialization;
+
+    using GitTools;
 
     [Serializable]
-    public class GitVersionException : ApplicationException
+    public class GitVersionException : GitToolsException
     {
-        public GitVersionException()
-        {
-        }
-
-
         public GitVersionException(string message)
             : base(message)
         {
@@ -19,12 +15,6 @@
 
         public GitVersionException(string message, Exception innerException)
             : base(message, innerException)
-        {
-        }
-
-
-        protected GitVersionException(SerializationInfo info, StreamingContext context)
-            : base(info, context)
         {
         }
     }

--- a/src/GitVersionCore/VersionCalculation/FallbackBaseVersionStrategy.cs
+++ b/src/GitVersionCore/VersionCalculation/FallbackBaseVersionStrategy.cs
@@ -14,11 +14,22 @@
     {
         public override IEnumerable<BaseVersion> GetVersions(GitVersionContext context)
         {
-            var baseVersionSource = context.Repository.Commits.QueryBy(new CommitFilter
+            Commit baseVersionSource;
+            var currentBranchTip = context.CurrentBranch.Tip;
+
+            try
             {
-                IncludeReachableFrom = context.CurrentBranch.Tip
-            }).First(c => !c.Parents.Any());
-            yield return new BaseVersion(context, "Fallback base version", false, new SemanticVersion(minor: 1), baseVersionSource, null);
+                baseVersionSource = context.Repository.Commits.QueryBy(new CommitFilter
+                {
+                    IncludeReachableFrom = currentBranchTip
+                }).First(c => !c.Parents.Any());
+            }
+            catch (NotFoundException exception)
+            {
+                throw new GitVersionException($"Can't find commit {currentBranchTip.Sha}. Please ensure that the repository is an unshallow clone with `git fetch --unshallow`.", exception);
+            }
+
+            yield return new BaseVersion(context, "Fallback base version", false, new SemanticVersion(minor : 1), baseVersionSource, null);
         }
     }
 }

